### PR TITLE
Restructure Inputted Data

### PR DESCRIPTION
**Pull request by APJohns**
Move most of the configuration into a properties input object.
Clean up existing data structure.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,6 +2,66 @@
...
+      hasSearch: boolean;
+      // TBD: Server Side / Client Side / Restricted
+    };
+    sort: {
+      defaultSortedColumn: string;
+      // TBD: Server Side / Client Side / Restricted
+    };
+    hasSelectableRows: boolean;
+    statusIndicatorMapping: {
+      'complete': string;
```
Comment (kaseyllabelle) [inline README.md]: Can we organize these either alphabetically or by progress?

So...
complete
incomplete-primary
incomplete-secondary
in-progress
warning

Or...
incomplete-primary
incomplete-secondary
in-progress
warning
complete

File: README.md [hunk 2]
```diff
@@ -2,6 +2,66 @@
...
+      columnNumber: number,
+      target: string,
+    }, */
+    // TODO: figure out path for <a> routerLink
+    actions: {
+        element: string; // Expect 'a', or 'button'
+        text: string;
+        ariaLabel: string; // Can use ${key} syntax to insert values frmo the row corresponding to the given key
+        class: string;
+        href: string;
```
Comment (kaseyllabelle) [inline README.md]: Can we add a comment here saying href doesn't apply if element = button?

File: README.md [hunk 3]
```diff
@@ -2,6 +2,66 @@
...
+      ariaLabel: string,
+      action: string,
+      columnNumber: number,
+      target: string,
+    }, */
+    // TODO: figure out path for <a> routerLink
+    actions: {
+        element: string; // Expect 'a', or 'button'
+        text: string;
+        ariaLabel: string; // Can use ${key} syntax to insert values frmo the row corresponding to the given key
```
Comment (kaseyllabelle) [inline README.md]: Typo in comment

File: src/app/app.component.ts [hunk 4]
```diff
@@ -9,36 +9,54 @@ import { ColumnType } from './model/column';
...
+    search:  {
+      hasSearch: true
+      // TBD: Server Side / Client Side / Restricted
+    },
+    sort: {
+      defaultSortedColumn: 'phone'
+      // TBD: Server Side / Client Side / Restricted
+    },
+    hasSelectableRows: true,
+    // TODO: Organize this
```
Comment (kaseyllabelle) [inline src/app/app.component.ts]: Can we organize these in this pr?

File: src/app/table/table-footer/column-selector/column-selector.component.html [hunk 5]
```diff
@@ -13,12 +13,12 @@
       <input 
         type="checkbox" 
         [id]="item.key" 
-        [name]="item.displayName"
+        [name]="item.headerText"
```
Comment (kaseyllabelle) [inline src/app/table/table-footer/column-selector/column-selector.component.html]: Can we change `item` to `column` here to give more context?

File: src/app/table/table-footer/column-selector/column-selector.component.html [hunk 6]
```diff
@@ -13,12 +13,12 @@
...
         [disabled]="item.key === defaultSortColName"
```
Comment (kaseyllabelle) [inline src/app/table/table-footer/column-selector/column-selector.component.html]: Can we change Col to Column at some point so we're using consistent language?

File: src/app/table/table.component.html [hunk 7]
```diff
@@ -40,7 +40,7 @@
         </thead>
 
         <!-- Table Loading -->
-        <tbody *ngIf="isTableLoading; else loaded">
+        <tbody *ngIf="isDataLoading; else loaded">
```
Comment (kaseyllabelle) [inline src/app/table/table.component.html]: I like this change

File: src/app/table/table.component.ts [hunk 8]
```diff
@@ -74,12 +103,12 @@ export class TableComponent implements OnInit, OnChanges {
   ngOnInit() {
     const displayDensityName = localStorage.getItem('selectedDensity') || 'Comfortable';
     this.setDisplayDensity(displayDensityName);
-    this.sortColName = this.defaultSortColName;
+    this.sortColName = this.properties.sort.defaultSortedColumn;
```
Comment (kaseyllabelle) [inline src/app/table/table.component.ts]: Can we change Col to Column here too at some point?

File: src/app/table/table.component.ts [hunk 9]
```diff
@@ -8,26 +8,64 @@ import { Column, ColumnType } from '../model/column';
...
+      columnNumber: number,
+      target: string,
+    }, */
+    // TODO: figure out path for <a> routerLink
+    actions: {
+        element: string; // Expect 'a', or 'button'
+        text: string;
+        ariaLabel: string; // Can use ${key} syntax to insert values frmo the row corresponding to the given key
+        class: string;
+        href: string;
```
Comment (kaseyllabelle) [inline src/app/table/table.component.ts]: Same re: href + button

File: src/app/table/table.component.ts [hunk 10]
```diff
@@ -8,26 +8,64 @@ import { Column, ColumnType } from '../model/column';
...
+      hasSearch: boolean;
+      // TBD: Server Side / Client Side / Restricted
+    };
+    sort: {
+      defaultSortedColumn: string;
+      // TBD: Server Side / Client Side / Restricted
+    };
+    hasSelectableRows: boolean;
+    statusIndicatorMapping: {
+      'complete': string;
```
Comment (kaseyllabelle) [inline src/app/table/table.component.ts]: Same re: organizing

File: README.md (hunk 2 above)
Comment (APJohns) [reply README.md]: We won't have href when we make the update to pass in paths to actions and links. href will be removed as part of that work

File: src/app/table/table.component.ts [hunk 11]
```diff
@@ -8,26 +8,64 @@ import { Column, ColumnType } from '../model/column';
...
+      ariaLabel: string,
+      action: string,
+      columnNumber: number,
+      target: string,
+    }, */
+    // TODO: figure out path for <a> routerLink
+    actions: {
+        element: string; // Expect 'a', or 'button'
+        text: string;
+        ariaLabel: string; // Can use ${key} syntax to insert values frmo the row corresponding to the given key
```
Comment (kaseyllabelle) [inline src/app/table/table.component.ts]: Typo in 'from'

File: src/app/table/table.component.ts [hunk 12]
```diff
@@ -8,26 +8,64 @@ import { Column, ColumnType } from '../model/column';
...
+        class: string;
+        href: string;
```
Comment (kaseyllabelle) [inline src/app/table/table.component.ts]: Add comments to match what's in src/app/table/table-columns/actions/actions.component.ts

**Pull request merged**